Sources/Networking/Auth/OAuthClient.swift

```diff
@@ -204,26 +207,45 @@ final public class DefaultOAuthClient: OAuthClient {
     public func getTokens(policy: AuthTokensCachePolicy) async throws -> TokenContainer {
         let localTokenContainer = tokenStorage.tokenContainer
 
+        if policy != .local,
+           let localTokenContainer,
+            localTokenContainer.decodedRefreshToken.isExpired() {
+            // The refresh token is expired, the token is un-refreshable
+            throw OAuthClientError.refreshTokenExpired
+        }
+
         switch policy {
+
         case .local:
             guard let localTokenContainer else {
                 Logger.OAuthClient.debug("Tokens not found")
                 throw OAuthClientError.missingTokens
             }
             Logger.OAuthClient.debug("Local tokens found, expiry: \(localTokenContainer.decodedAccessToken.exp.value, privacy: .public)")
             return localTokenContainer
+
         case .localValid:
             guard let localTokenContainer else {
                 Logger.OAuthClient.debug("Tokens not found")
                 throw OAuthClientError.missingTokens
             }
-            Logger.OAuthClient.debug("Local tokens found, expiry: \(localTokenContainer.decodedAccessToken.exp.value, privacy: .public)")
-            if localTokenContainer.decodedAccessToken.isExpired() {
+            let tokenExpiryDate = localTokenContainer.decodedAccessToken.exp.value
+            Logger.OAuthClient.debug("Local tokens found, expiry: \(tokenExpiryDate, privacy: .public)")
+
+#if DEBUG
+            let expiresSoon = false
+#else
+            // Expires in less than 10 minutes, check only in release, the staging token expires every 4 minutes
```

Comment:
I've added a local check for the token expiry date, if it's less than 10 minutes we treat it as expired

Reply:
changed!